File: src/main/java/com/dozycoffee/wms/warehouse/application/port/in/DeactivateWarehouseUseCase.java
package com.dozycoffee.wms.warehouse.application.port.in;

import com.dozycoffee.wms.warehouse.application.port.in.result.WarehouseResult;
import reactor.core.publisher.Mono;

public interface DeactivateWarehouseUseCase {

    Mono<WarehouseResult> deactivate(Long warehouseId);
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/application/port/in/DeactivateWarehouseUseCase.kt
package com.dozycoffee.wms.warehouse.application.port.`in`

import com.dozycoffee.wms.warehouse.application.port.`in`.result.WarehouseResult
import reactor.core.publisher.Mono

interface DeactivateWarehouseUseCase {
    fun deactivate(warehouseId: Long): Mono<WarehouseResult>
}
